README.md

```diff
@@ -22,15 +201,78 @@ docker build --build-arg OPENJDK_VERSION=11 .
 
 ### Unit testing
 
-- **TEST_GITHUB_TOKEN** env variable is needed
+- **TEST_GITHUB_TOKEN** env variable is needed.
 
 ### Integration testing
 
 In order to execute integration testing you just run `env GITHUB_TOKEN=%TOKEN% URL=%GITHUB_EVENT_URL% ['parent-dependencies=%PARENT_DEPENDENCIES%'] ['child-dependencies=%CHILD_DEPENDENCIES%'] yarn it` where:
 
-- %TOKEN%: is your personal token, like `1e2ca1ac1252121d83fbe69ab3c4dd92bcb1ae32`
-- %GITHUB_EVENT_URL%: the url to your event to test, like `https://github.com/kiegroup/lienzo-core/pull/3`
-- %PARENT_DEPENDENCIES%: the OPTIONAL comma separated parent project list
-- %CHILD_DEPENDENCIES%: the OPTIONAL comma separated child project list
+- %TOKEN%: is your personal token, like `1e2ca1ac1252121d83fbe69ab3c4dd92bcb1ae32`.
+- %GITHUB_EVENT_URL%: the url to your event to test, like `https://github.com/kiegroup/lienzo-core/pull/3`.
+- %PARENT_DEPENDENCIES%: the OPTIONAL comma separated parent project list.
+- %CHILD_DEPENDENCIES%: the OPTIONAL comma separated child project list.
+
+So the final command would look like `env GITHUB_TOKEN=3e6ce1ac1772121d83fbe69ab3c4dd92dad1ae40 URL=https://github.com/kiegroup/lienzo-core/pull/3 'parent-dependencies=lienzo-core,lienzo-tests' 'child-dependencies=appformer' yarn it`.
+
+## Github limitations
+
+### matrix in uses
+
+It's not possible to use matrix variables in `uses` like this:
+
+```
+jobs:
+  build-chain-jdk:
+    strategy:
+      matrix:
+        buid-chain-system: [openjdk8, openjdk11]
+    runs-on: ubuntu-latest  
+    name: Pull Request  
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build Chain
+        id: build-chain
+        uses: kiegroup/github-action-build-chain@${{ matrix.buid-chain-system }}
+        with:
+          ...
+          ...
+```
+
+instead you have to duplicate jobs for different environments, like:
+```
+jobs:
+  build-chain-openjdk8:
+    runs-on: ubuntu-latest
+    name: Build Pull Request openjdk8
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build Chain
+      id: build-chain
+      uses: kiegroup/github-action-build-chain@openjdk8
+      with:
+        ...
+        ...
+  build-chain-openjdk11:
+    runs-on: ubuntu-latest
+    name: Build Pull Request openjdk11
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build Chain
+      id: build-chain
+      uses: kiegroup/github-action-build-chain@openjdk11
+      with:
+        ...
+        ...  
+```
+
+### workflow-file-name
+
+You are probably wondering why the input field `workflow-file-name` even exists. Why don't we take the filename directly from the job and keep the same name for all the flow files in the chain?. Well... we can in case the `name` is not defined in the flow, then the file name information can be taken from `GITHUB_WORKFLOW` environment variable but in case the name is set, `GITHUB_WORKFLOW` becomes the name and there's no other way to get filename from the tool.
```

Comment:
```suggestion
You are probably wondering why the input field `workflow-file-name` even exists. Why don't we take the filename directly from the job and keep the same name for all the flow files in the chain?. Well, we can in case the `name` is not defined in the flow, then the file name information can be taken from `GITHUB_WORKFLOW` environment variable but in case the name is set, `GITHUB_WORKFLOW` becomes the name and there's no other way to get filename from the tool.
```